ocaml/xapi/xapi_vm_helpers.ml

```diff
@@ -772,49 +773,30 @@ let choose_host_uses_wlb ~__context =
       (Db.Pool.get_other_config ~__context
          ~self:(Helpers.get_pool ~__context)))
 
-
-(* This is a stub used in the pattern_matching below to silence a
- * warning in the newer ocaml compilers *)
-exception Float_of_string_failure
-
 (** Given a virtual machine, returns a host it can boot on, giving   *)
 (** priority to an affinity host if one is present. WARNING: called  *)
 (** while holding the global lock from the message forwarding layer. *)
 let choose_host_for_vm ~__context ~vm ~snapshot =
   if choose_host_uses_wlb ~__context then
     try
-      let rec filter_and_convert recs =
-        match recs with
-        | (h, recom) :: tl ->
-          begin
-            debug "\n%s\n" (String.concat ";" recom);
-            match recom with
-            | ["WLB"; "0.0"; rec_id; zero_reason] ->
-              filter_and_convert tl
-            | ["WLB"; stars; rec_id] ->
-              let st = try float_of_string stars with Failure _ -> raise Float_of_string_failure
-              in
-              (h, st, rec_id) :: filter_and_convert tl
-            | _ -> filter_and_convert tl
-          end
-        | [] -> []
+      let possible_rec (host, recommendation) =
+        match recommendation with
+        | Recommendation {source; id; score} ->
+            Some (host, score, id)
+        | _ ->
+            None
       in
+      let all_hosts = retrieve_wlb_recommendations ~__context ~vm ~snapshot
+        |> List.filter_map possible_rec
+        |> List.sort (fun (_, s, _) (_, s', _) -> Float.compare s' s)
```

Comment:
Don't you need to reverse the list here? The previous comparison function sorted high values first, this one sorts low values first.

Reply:
`s` and `s'` are reversed to have high values first:

```
# List.sort (fun s s' -> Float.compare s' s) [1.; 2.; 3.];;
- : float list = [3.; 2.; 1.]
```

Reply:
can you add a comment about that? (or name the function something like `cmp_descending`)